gensim/matutils.py

```diff
@@ -141,8 +141,7 @@ def corpus2csc(corpus, num_terms=None, dtype=np.float64, num_docs=None, num_nnz=
             if printprogress and docno % printprogress == 0:
                 logger.info("PROGRESS: at document #%i/%i", docno, num_docs)
             posnext = posnow + len(doc)
-            indices[posnow: posnext] = [feature_id for feature_id, _ in doc]
-            data[posnow: posnext] = [feature_weight for _, feature_weight in doc]
+            indices[posnow: posnext], data[posnow: posnext] = zip(*doc)
```

Comment:
-1 on all the changes in this commit: `zip(*)` fails on empty docs, and is harder to read.

Reply:
But it makes it twice as fast! I can add a comment and a check for the document length.

Reply:
OK… let's see the cleaned up code with comments.

IMO this method should be cythonized for a proper speed up. Not sure if obfuscation is the way to go here. CC @mpenkov